### PR TITLE
Add flake8 and black checks on pull requests

### PR DESCRIPTION
**Pull request by sityaevDI**
modified setup.py and removed unused libs
added flake8 to pre-commit
added flake8 step to github actions, but I can't test it locally.

### CODE REVIEW

**Comment by sityaevDI** [comment]
I think it's better to create another CI file and split the logic for pull request and push workflows. Thoughts?

**Comment by sityaevDI** [comment]
please check [this pull request](https://github.com/cdisc-org/cdisc-rules-engine/pull/87), maybe it's the better way to introduce checks

File: .github/workflows/automated-ci.yml [hunk 1]
```diff
@@ -15,17 +15,60 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
```
Comment (alexfurmenkov) [inline .github/workflows/automated-ci.yml:19]: What is the purpose of this step?

File: .github/workflows/automated-ci.yml [hunk 2]
```diff
@@ -15,17 +15,60 @@ jobs:
...
           pip install -r requirements.txt
-      
-    # running the tests
+
+      # running the tests
       - name: Running Tests
         run: |
           python -m pytest tests/unit/
+
+  changed_files:
```
Comment (alexfurmenkov) [inline .github/workflows/automated-ci.yml:36]: Let's rename it to be like `get_changed_files`

File: .github/workflows/automated-ci.yml (hunk 1 above)
Comment (sityaevDI) [reply .github/workflows/automated-ci.yml:19]: it was useful for fetching more data on getting diffs on push event, forgot to remove.

**Review by alexfurmenkov — APPROVED** [review]

**Pull request merged**